File: Python/Enums/messageCode.py
from enum import Enum

class MessageCodes(Enum):
    """
    eStation message codes
    """
    Ok = 0
    Idle = 1
    Result = 2
    Heartbeat = 3
    ModError = 4
    AppEror = 5
    Busy = 6
    MaxLimit = 7
    Invalid_Task_ESL = 8
    Invalid_Task_DSL = 9
    Invalid_Config = 10
    Invalid_OTA = 11
    Invalid_Firmware = 12

File: Java/src/main/java/com/estation/demo/enums/MessageCode.java
package com.estation.demo.enums;

public enum MessageCode {
    OK(0),
    IDLE(1),
    RESULT(2),
    HEARTBEAT(3),
    MOD_ERROR(4),
    APP_ERROR(5),
    BUSY(6),
    MAX_LIMIT(7),
    INVALID_TASK_ESL(8),
    INVALID_TASK_DSL(9),
    INVALID_CONFIG(10),
    INVALID_OTA(11),
    INVALID_FIRMWARE(12);

    private final int value;

    MessageCode(int value) {
        this.value = value;
    }

    public int value() {
        return value;
    }

    public static MessageCode fromValue(int value) {
        for (MessageCode code : values()) {
            if (code.value == value) {
                return code;
            }
        }
        throw new IllegalArgumentException("Unknown MessageCode value: " + value);
    }
}
